### PR TITLE
FrenzyConfig

### PR DESCRIPTION
**Pull request by vsoneji**
Apply frenzyConfig into FrenzyBot
- existing values are overwritten in a method
- duck loop rearranged to use frenzyConfig directly
- logic added to slow down the first or 'auto' duck
- added TODOs for missing config fields

### CODE REVIEW

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/bots/FrenzyBot.java [hunk 1]
```diff
@@ -104,6 +155,7 @@ public void init(LinearOpMode owner, HardwareMap ahwMap, Telemetry telemetry) th
 
         try {
             this.frenzyConfig = this.botConfig.getFreightFrenzyConfig();
+            this.applyFrenzyConfig();
```
Comment (sashajeltuhin) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/bots/FrenzyBot.java:158]: It is too early to ask for the config in the constructor. It is processed in the init() method

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/bots/FrenzyBot.java [hunk 2]
```diff
@@ -86,8 +92,53 @@
     private static double INTAKE_SPEED = -0.3;
     private static double INTAKE_SPEED_REVERSE = 0.3;
 
+    private boolean firstDuckDropped = false;
     private boolean intakeRunning = false;
 
+    private void applyFrenzyConfig() {
+        // Dropper
+        DROPPER_SERVO_POS_DROP = frenzyConfig.getDropperPositionDrop();
```
Comment (sashajeltuhin) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/bots/FrenzyBot.java:100]: Should we check the incoming numbers for validity? What if the file is corrupt or empty? What is the best fail-safety mechanism, e.g. check for 0 values and use hard-coded numbers instead?

**Review by sashajeltuhin — CHANGES_REQUESTED** [review]

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/bots/FrenzyBot.java (hunk 2 above)
Comment (vsoneji) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/bots/FrenzyBot.java:100]: as we move forward, the java code values will progressively get more and more outdated.
If we fall back on the hard-coded numbers, the robot's behavior will be VERY unpredictable.

We could put a process in place where we periodically update the hard-coded values from the JSON file. But I'd much rather have robot fail **completely** if the JSON is missing.

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/bots/FrenzyBot.java (hunk 1 above)
Comment (vsoneji) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/bots/FrenzyBot.java:158]: super.init() is already called by this time, so the 'botConfig' object should be populated by this time, isn't it?

**Pull request closed without merging**